question25/main.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const textLinkToTerms = document.getElementById('js-terms-textlink');
+const modal = document.getElementById('js-modal');
+const modalClose = document.getElementById('js-close');
+const mask = document.getElementById('js-mask');
+const checkbox = document.getElementById('js-checkbox');
+const submitButton = document.getElementById('js-submit-button');
+const modalContainer = document.getElementById("js-modal-container");
+const name = document.getElementById('name');
+const mail = document.getElementById('mail');
+const password = document.getElementById('password');
+
+
+textLinkToTerms.addEventListener('click',()=>{
+    modal.classList.remove('hidden');
+    mask.classList.remove('hidden');
+});
+modalClose.addEventListener('click',()=>{
+    modal.classList.add('hidden');
+    mask.classList.add('hidden');
+});
+mask.addEventListener('click',()=>{
+    modalClose.click();
+});
+
+const options = {
+  root: document.getElementById('js-modal'),
+  threshold: 1
+};
+
+const checkWhenIntersect = ([entry]) => {
+    if (entry.isIntersecting) {
+      checkbox.checked = true;
+      checkbox.disabled = false;
+      switchSubmitButton(); 
+    }
+}
+
+const observer = new IntersectionObserver(checkWhenIntersect, options);
+observer.observe(modalContainer.lastElementChild);
+
+
+
+const validationInfo = {
+  name: {
+    status: false,
```

Comment:
IMO
I found it difficult to understand what the status represents.
How about naming it something more specific?

Reply:
Thank you for your review! 
I changed the name to a more appropriate name following the teacher's opinion.

https://github.com/csakyo/homework/pull/39#discussion_r945218383
Fixed commit : https://github.com/csakyo/homework/pull/39/commits/a11013fb15bc8dfd7bdbc8fc0f8800d809cd699a